tree/util.go

```diff
@@ -142,17 +142,17 @@ func CommitTrees(
 		return err
 	}
 
-	for idx := range *assetTrees {
-		err := func(i int) error {
+	for idx := int64(0); idx < assetTrees.Size(); idx++ {
```

Comment:
Please come up with a solution to avoid iterat `assetTrees`, the size can be huge, we should get a more small scope about which AccountAsset are needed to commit.